services/director-v2/src/simcore_service_director_v2/modules/dynamic_sidecar/monitor/models.py

```diff
@@ -0,0 +1,393 @@
+import datetime
+import json
+import logging
+from enum import Enum
+from typing import Any, Dict, List, Optional
+from uuid import UUID
+
+from models_library.basic_regex import VERSION_RE
+from models_library.projects import ProjectID
+from models_library.projects_nodes import NodeID
+from models_library.service_settings_labels import (
+    ComposeSpecLabel,
+    PathsMappingLabel,
+    SimcoreServiceLabels,
+)
+from models_library.services import SERVICE_KEY_RE
+from pydantic import BaseModel, Field, PositiveInt, validator
+
+from ....models.domains.dynamic_services import DynamicServiceCreate
+from ....models.schemas.constants import UserID
+from .utils import AsyncResourceLock
+
+logger = logging.getLogger()
+
+
+class DynamicSidecarStatus(str, Enum):
+    OK = "ok"  # running as expected
+    FAILING = "failing"  # requests to the sidecar API are failing service should be cosnidered as unavailable
+
+
+class OverallStatus(BaseModel):
+    """Generated from data from docker container inspect API"""
+
+    status: DynamicSidecarStatus = Field(..., description="status of the service")
+    info: str = Field(..., description="additional information for the user")
+
+    def _update(self, new_status: DynamicSidecarStatus, new_info: str) -> None:
+        self.status = new_status
+        self.info = new_info
+
+    def update_ok_status(self, info: str) -> None:
+        self._update(DynamicSidecarStatus.OK, info)
+
+    def update_failing_status(self, info: str) -> None:
+        self._update(DynamicSidecarStatus.FAILING, info)
+
+    def __eq__(self, other: "OverallStatus") -> bool:
+        return self.status == other.status and self.info == other.info
+
+    @classmethod
+    def make_initially_ok(cls) -> "OverallStatus":
+        # the service is initially ok when started
+        initial_state = cls(status=DynamicSidecarStatus.OK, info="")
+        return initial_state
+
+
+class DockerStatus(str, Enum):
+    CREATED = "created"
+    RUNNING = "running"
+    PAUSED = "paused"
+    RESTARTING = "restarting"
+    REMOVING = "removing"
+    EXITED = "exited"
+    DEAD = "dead"
+
+
+class DockerContainerInspect(BaseModel):
+    status: DockerStatus = Field(
+        ...,
+        scription="status of the underlying container",
+    )
+    name: str = Field(..., description="docker name of the container")
+    id: str = Field(..., description="docker id of the container")
+
+    last_updated: datetime.datetime = Field(
+        default_factory=datetime.datetime.utcnow,
+        description="time of the update in UTC",
+    )
+
+    @classmethod
+    def from_container(cls, container: Dict[str, Any]) -> "DockerContainerInspect":
+        return cls(
+            status=DockerStatus(container["State"]["Status"]),
+            name=container["Name"],
+            id=container["Id"],
+        )
+
+
+class DynamicSidecar(BaseModel):
+    overall_status: OverallStatus = Field(
```

Comment:
consider just calling it status. especially since there is no other status anyway,